pkg/remote/fileshare/s3fs/s3fs.go

```
@@ -31,94 +48,839 @@ func NewS3Client(config *aws.Config) *S3Client {
+	if objectKey == "" || objectKey == "/" {
+		_, err := c.client.HeadBucket(ctx, &s3.HeadBucketInput{
+			Bucket: aws.String(bucketName),
+		})
+		exists := true
+		if err != nil {
+			var apiError smithy.APIError
+			if errors.As(err, &apiError) {
+				switch apiError.(type) {
+				case *types.NotFound:
+					exists = false
+					err = nil
```

golangci-lint (1.62.2)

ineffectual assignment to err

(ineffassign)